getVfRep.py

```diff
@@ -0,0 +1,98 @@
+from common import TFT_TOOLS_IMG, PluginOutput, j2_render, TftAggregateOutput
+from logger import logger
+from testConfig import TestConfig
+from iperf import IperfServer, IperfClient
+from thread import ReturnValueThread
+from task import Task
+from host import Result
+from typing import Optional, Union
+import sys
+import yaml
+import json
+import jc
+
+class GetVfRep(Task):
+    def __init__(self, tft: TestConfig, iperf_instance: Union[IperfServer, IperfClient], tenant: bool):
+        super().__init__(tft, 0, iperf_instance.node_name, tenant, True)
+
+        self.in_file_template = "./manifests/tools-pod.yaml.j2"
+        self.out_file_yaml = f"./manifests/yamls/tools-pod-{self.node_name}-get-vf-rep.yaml"
+        self.template_args["pod_name"] = f"tools-pod-{self.node_name}-get-vf-rep"
+        self.template_args["test_image"] = TFT_TOOLS_IMG
+
+        self.pod_name = self.template_args["pod_name"]
+        self.iperf_pod_name = iperf_instance.template_args["pod_name"]
+
+        j2_render(self.in_file_template, self.out_file_yaml, self.template_args)
+        logger.info(f"Generated Server Pod Yaml {self.out_file_yaml}")
+        
+
+    def extract_vf_rep(self) -> str:
+        get_vf_rep_cmd = f'exec -n default {self.pod_name} -- /bin/sh -c "crictl --runtime-endpoint=/host/run/crio/crio.sock ps -a --name={self.iperf_pod_name} -o json "'
+        vf_rep_json = self.run_oc(get_vf_rep_cmd).out
+        data = json.loads(vf_rep_json)
+        return data["containers"][0]["podSandboxId"][:15]
+
+    def run_ethtool_cmd(self, vf_rep: str) -> (int, int):
+        self.ethtool_cmd = f'exec -n default {self.pod_name} -- /bin/sh -c "ethtool -S {vf_rep}"'
+        ethtool_output = self.run_oc(self.ethtool_cmd).out
+        #print("DEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEBAAAAAAAAAAAAAG %s" % ethtool_output)
+        rxpacket = self.parse_out_packet(ethtool_output, "rx_packet")
```

Comment:
As discussed please add a delay/sleep here for now.

Please look into semaphores and barriers for synchronization techniques between different running processes.

Reply:
I can add a delay/sleep but just to clarify all these commands are currently being run synchronously on a single thread. So because this is running on a single thread I'm curious as to how we will use semaphores?

Reply:
There are multiple threads running at once i.e. the actual iperf flow tests, the other plugins. When we run task.start() on each task, these tasks will run concurrently.

For now we can just add a sleep so that your plugin will run roughly the duration of the iperf flow test. However, a better way to do this that we will need to figure out for all plugins is to synchronize collecting plugin data with the internals of the iperf test. This would require some sort of semaphore or barrier to ensure that T1 in one task corresponds to T1 in every other tasks.